### PR TITLE
Clang-Tidy: use modern for each loops and transparent functors

### PR DESCRIPTION
**Pull request by Skylion007**
This applies some more clang-tidy fixups. Particularly, this applies the modernize loops and modernize-use-transparent-functors checks. Transparent functors are less error prone since you don't have to worry about accidentally specifying the wrong type and are newly available as of C++17.

Modern foreach loops tend be more readable and can be more efficient to iterate over since the loop condition is removed.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @ezyang

### CODE REVIEW

**Comment by Skylion007** [comment]
Sigh, another flake. This check is failing on master too right now.

File: aten/src/ATen/native/nested/NestedTensorUtils.h [hunk 1]
```diff
@@ -384,8 +384,8 @@ inline Tensor wrap_tensor_node(
     long sizes_offset = 0;
     for (size_t i = 0; i < tensor_node.degree(); ++i) {
       auto tensor_sizes = tensor_node.children(i).sizes();
-      for (size_t j = 0; j < tensor_sizes.size(); ++j) {
-        nt_sizes.data_ptr<int64_t>()[sizes_offset++] = tensor_sizes[j];
+      for (long tensor_size : tensor_sizes) {
```
Comment (ezyang) [inline aten/src/ATen/native/nested/NestedTensorUtils.h:387]: seems better to use `int64_t` here

File: torch/csrc/autograd/profiler_python.cpp [hunk 2]
```diff
@@ -933,7 +933,7 @@ class PostProcess {
   template <EventType E>
   struct State {
     ska::flat_hash_map<TraceKey, ExtraFields<E>> fields_;
-    std::priority_queue<Exit, std::vector<Exit>, std::greater<Exit>> exits_;
+    std::priority_queue<Exit, std::vector<Exit>, std::greater<>> exits_;
```
Comment (ezyang) [inline torch/csrc/autograd/profiler_python.cpp:936]: wow such deduction (elementary my dear watson)

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**